lib/ansible/modules/user.py

```diff
@@ -1372,14 +1372,25 @@ def create_homedir(self, path):
             if os.path.exists(self.LOGIN_DEFS):
                 with open(self.LOGIN_DEFS, 'r') as f:
                     for line in f:
+                        # HOME_MODE has higher precedence as UMASK
+                        m = re.match(r'^HOME_MODE\s+(\d+)$', line)
+                        if m:
+                            mode = int(m.group(1), 8)
+                            break  # higher precedence
                         m = re.match(r'^UMASK\s+(\d+)$', line)
                         if m:
                             umask = int(m.group(1), 8)
                             mode = 0o777 & ~umask
-                            try:
-                                os.chmod(path, mode)
-                            except OSError as e:
-                                self.module.exit_json(failed=True, msg="%s" % to_native(e))
+                # fallback if neither HOME_MODE nor UMASK are set;
+                # follow behaviour of useradd initializing UMASK = 022
+                try:
```

Comment:
How about 

1. setting `mode` to `None` initially and checking it here 

```python
    if mode is None:
        mode = 0o755
```

or 

2. Setting `mode` to `0o755` by default

Reply:
Good point, that's a cleaner approach. Will do.